micromanager_gui/main_window.py

```diff
@@ -187,41 +143,166 @@ def __init__(
-        self.crop_Button.setEnabled(enabled)
+        self.obj.objective_groupBox.setEnabled(enabled)
+        self.ill.illumination_Button.setEnabled(enabled)
+        self.tab.tabWidget.setEnabled(enabled)
 
-    def _update_exp(self, exposure: float):
-        self._mmc.setExposure(exposure)
-        if self.streaming_timer:
-            self.streaming_timer.setInterval(int(exposure))
-            self._mmc.stopSequenceAcquisition()
-            self._mmc.startContinuousSequenceAcquisition(exposure)
+        self.mda._set_enabled(enabled)
+        if self._mmc.getXYStageDevice():
+            self.explorer._set_enabled(enabled)
+        else:
+            self.explorer._set_enabled(False)
 
-    def _on_exp_change(self, camera: str, exposure: float):
-        with blockSignals(self.exp_spinBox):
-            self.exp_spinBox.setValue(exposure)
-        if self.streaming_timer:
-            self.streaming_timer.setInterval(int(exposure))
+    def browse_cfg(self):
+        self._mmc.unloadAllDevices()  # unload all devicies
+
+        self._set_enabled(False)
+
+        # clear spinbox/combobox without accidently setting properties
+        boxes = [
+            self.obj.objective_comboBox,
+            self.cam.bin_comboBox,
+            self.cam.bit_comboBox,
+            self.tab.snap_channel_comboBox,
+            self.stages.xy_device_comboBox,
+            self.stages.focus_device_comboBox,
+        ]
+        with blockSignals(boxes):
+            for box in boxes:
+                box.clear()
+
+        self.mda.clear_channel()
+        self.mda.clear_positions()
+        self.explorer.clear_channel()
+
+        self.objectives_device = None
+        self.objectives_cfg = None
+
+        file_dir = QtW.QFileDialog.getOpenFileName(self, "", "", "cfg(*.cfg)")
+        self.cfg.cfg_LineEdit.setText(str(file_dir[0]))
+        self.tab.max_min_val_label.setText("None")
+        self.cfg.load_cfg_Button.setEnabled(True)
+
+    def load_cfg(self):
+        self.cfg.load_cfg_Button.setEnabled(False)
+        cfg = self.cfg.cfg_LineEdit.text()
+        if cfg == "":
+            cfg = "MMConfig_demo.cfg"
+            self.cfg.cfg_LineEdit.setText(cfg)
+        self._mmc.loadSystemConfiguration(cfg)
+        self._refresh_options()
+        self._set_enabled(True)
+
+    def _refresh_options(self):
+        self._refresh_camera_options()
+        self._refresh_objective_options()
+        self._refresh_channel_list()
+        self._refresh_positions()
+        self._refresh_xyz_devices()
+
+    def update_viewer(self, data=None):
+        if data is None:
+            try:
+                data = self._mmc.getLastImage()
+            except (RuntimeError, IndexError):
+                # circular buffer empty
+                return
+        try:
+            preview_layer = self.viewer.layers["preview"]
+            preview_layer.data = data
+        except KeyError:
+            preview_layer = self.viewer.add_image(data, name="preview")
+
```

Comment:
is there any changes in this whole section? I'm not sure why this is showing up on github, but a lot of this I think is showing up due to how you pulled in new changes (as commits rather than merge commits or rebases)

Reply:
I think this is just because I reordered the methods